homeassistant/config_entries.py

```diff
@@ -817,13 +817,19 @@ def async_get_options_flow(config_entry: ConfigEntry) -> "OptionsFlow":
         raise data_entry_flow.UnknownHandler
 
     @callback
-    def _abort_if_unique_id_configured(self) -> None:
+    def _abort_if_unique_id_configured(self, updates: Dict[Any, Any] = None) -> None:
         """Abort if the unique ID is already configured."""
+        assert self.hass
         if self.unique_id is None:
             return
 
-        if self.unique_id in self._async_current_ids():
-            raise data_entry_flow.AbortFlow("already_configured")
+        for entry in self._async_current_entries():
+            if entry.unique_id == self.unique_id:
+                if updates is not None and not updates.items() <= entry.data.items():
```

Comment:
Let's add a test that we don't update anything if updates are contained in `entry.data`. Patch `self.hass.config_entries.async_update_entry` and make sure it is not called.

Reply:
Added.